app/src/main/java/samples/despotoski/nikola/com/bottomsheetsample/BottomSheetActivity.kt

```diff
@@ -0,0 +1,124 @@
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.android.synthetic.main.bottom_sheet_1.view.*
+import samples.despotoski.nikola.com.bottomsheetsample.databinding.ActivityBottomSheetBinding
+import java.util.*
+
+class BottomSheetActivity : AppCompatActivity() {
+    private var mBottomSheetBehavior: BottomSheetBehavior<*>? = null
+
+    lateinit var binding : ActivityBottomSheetBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = DataBindingUtil.setContentView(this,R.layout.activity_bottom_sheet)
+
+        setSupportActionBar(binding.toolbar)
+
+        setupBottomSheet1()
+
+        setupBottomSheet2()
+    }
+
+    private fun setupBottomSheet2() {
+        binding.showBottomSheet2.setOnClickListener {
+            val bottomSheetDialogFragment: BottomSheetDialogFragment = CustomBottomSheetDialogFragment()
+            bottomSheetDialogFragment.show(supportFragmentManager, bottomSheetDialogFragment.tag)
+        }
+    }
+
+    private fun setupBottomSheet1() {
+        binding.showBottomSheet1.setOnClickListener { //Let's peek it, programmatically
+            mBottomSheetBehavior?.peekHeight = binding.layoutBottomSheet1.textViewDragMe.height
+            binding.layoutBottomSheet1.textViewDragMe.requestLayout()
+        }
+
+        val recyclerView = binding.layoutBottomSheet1.recyclerview
+        recyclerView.layoutManager = LinearLayoutManager(this).apply {
+            isAutoMeasureEnabled = true
+        }
+        recyclerView.adapter = ApplicationAdapter(this, listApplications(this))
+        val parentThatHasBottomSheetBehavior = recyclerView.parent.parent as FrameLayout
+        mBottomSheetBehavior = BottomSheetBehavior.from(parentThatHasBottomSheetBehavior)
+        mBottomSheetBehavior?.let {
+            binding.stateText.setText(getStateAsString(it.state))
+            it.setBottomSheetCallback(object : BottomSheetCallback() {
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    binding.stateText.setText(getStateAsString(newState))
+                }
+
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                    binding.offsetText.text = getString(R.string.offset, slideOffset)
+                }
+            })
+        }
+    }
+
+    override fun onBackPressed() {
+        mBottomSheetBehavior?.let {
```

Comment:
Since it's in Kotlin, perhaps remove the `m` prefix?

Reply:
done